content/general/concepts/algorithm/terms/selection-sort/selection-sort.md

```diff
@@ -0,0 +1,76 @@
+---
+Title: 'Selection-Sort Algorithm'
+Description: 'Returns a sorted array'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Algorithms'
+  - 'Arithmetic'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**Selection sort** is an in-place comparison sorting algorithm. It divides the input array into two parts: the sorted subarray at the beginning and the unsorted subarray at the end. The algorithm repeatedly selects the minimum (or maximum) element from the unsorted subarray and swaps it with the first unsorted element, expanding the sorted subarray by one element.
+
+## Explanation
+
+A brief explanation of `Selection Sort`:
+
+- Selection sort is an in-place comparison sorting algorithm: This means that selection sort does not require additional memory beyond the input array, and elements are compared to determine their relative order.
+
+- It divides the input array into two parts: Conceptually, selection sort divides the array into two subarrays: the sorted subarray at the beginning and the unsorted subarray at the end. Initially, the sorted subarray is empty, and the unsorted subarray contains all the elements.
+
+- The algorithm repeatedly selects the minimum (or maximum) element: In each iteration, selection sort identifies the minimum (or maximum) element from the unsorted subarray. The selection process involves scanning the unsorted subarray to find the smallest (or largest) element.
+
+- It swaps the selected element with the first unsorted element: Once the minimum (or maximum) element is identified, selection sort swaps it with the first element of the unsorted subarray. This action effectively expands the sorted subarray by one element and reduces the unsorted subarray's size.
+
+- The process continues until the entire array is sorted: Selection sort repeats the selection and swapping steps for the remaining unsorted elements until the entire array becomes sorted. The sorted subarray gradually grows, and the unsorted subarray shrinks until no elements remain in the unsorted portion.
+
+**Graphical representation of Selection Sort:**
+
+![Selection Sort](SelectionSort.png)
```

Comment:
You need to move the image to the folder below and use this address in the parentheses

https://raw.githubusercontent.com/Codecademy/docs/main/media/SelectionSort.png